### ISSUE TITLE
 Change text and appearence for start page of Life Version

### ISSUE THREAD

**Issue by mfranke93** [UniStuttgart-VISUS/damast#7]
*Initially opened by gs108488 in [222@TIK](https://github.tik.uni-stuttgart.de/frankemx/damast/issues/222). Relevant history:*

**gs108488** Dec 16, 2021:

> For the Life Version, the start page needs a complete make over. Apart from "English only" (see ~~[\#210](https://github.tik.uni-stuttgart.de/frankemx/damast/issues/210)~~), the start page will need different sections. For the design, the styles inherited from the overall base.htm need to be considered.

**Comment by tutebatti** [comment]
Is there an easy possibility to provide a kind of "sandbox" for me to check how my drafts for the startpage look inside the `main`-element?

**Comment by mfranke93** [comment]
The easiest way would be that you set up the development environment on your end. The instructions for that are all in the README.md, but if you get stuck, I could also help you set it up via video call.
________________________________________

**Comment by tutebatti** [comment]
The principal design is now there from our side. Maybe it makes sense that you, @mfranke93, have a brief look at it if it can be generally implemented that way. I would not want to upload these drafts, I will send them via mail.

The content will change, so this is just a preliminary check.

**Comment by mfranke93** [comment]
I am unsure how exactly you want to handle this because of the bimodal communication, email and issue.

For now, I will send you feedback via mail. Let me know if it is okay for you to continue the discussion here, but in that case, it would be a good idea to have the designs here as well for context.

**Comment by tutebatti** [comment]
We can certainly move it here after a first check-up. On the other hand, changes of content will take place through corrections and discussions with people outside github. So it is debatable at what point we should integrate this.

**Comment by mfranke93** [comment]
I have started incorporating everything in the [`feature/new-home-page`](https://github.com/UniStuttgart-VISUS/damast/tree/feature/new-home-page) branch now. This is still work in progress, though. A few notes:

 * As long as the copyright questions are unresolved, I have added a placeholder image for the background, and linked to the external resources for the logos.
 * I moved the explicit `<section>`s for the parallax effect to the `background-image` of the `<main>` element. The individual children of the `<main>` instead have a white background.
 * The direct children of the `<main>` are now `<article>` elements (from `<div class="text_section">`. The `<div class="text_subsection">` are now `<section class="subsection">`. Avoid using `<div>`.
 * Cleaned up the markup and CSS.
 * Changed `snake_case` IDs and class names to `kebab-case` to align with rest of code base and best practice.
 * Added an `<aside>` element for the in-page navigation, such that it is always visible.

There are other changes as well, but those should be the most critical ones for you to continue working on the content of the start page independently of the styling.

**Comment by tutebatti** [comment]
We have the ok to use the picture. However, I am not sure about the license. (I could ask, but I doubt that we can share it under the same license as the rest of the software.) Is it possible to exclude the picture from the rest, either legally via a disclaimer or technically by linking to an external resource?

**Comment by tutebatti** [comment]
@mfranke93 Any thoughts on what file format, resolution, ratio, etc. should be used regarding the screenshots for the startpage?

**Comment by mfranke93** [comment]
> We have the ok to use the picture. However, I am not sure about the license. (I could ask, but I doubt that we can share it under the same license as the rest of the software.) @mfranke93 Is it possible to exclude the picture from the rest, either legally via a disclaimer or technically by linking to an external resource?

Not sure what you mean with "via a disclaimer", but if it cannot be published under the MIT license, it absolutely *should not* be in the repository. In that case, we can either (1) bake it into the server software during deployment, or (2) directly link to an external URL for the image. The second one is what we already do for the university logos, and it would be my preference for the background image as well.

> @mfranke93 Any thoughts on what file format, resolution, ratio, etc. should be used regarding the screenshots for the startpage?

I would suggest PNG for the file format. Ratio should not matter, maybe not too extreme (under 2:1 or 1:2). Resolution: matching roughly the size it is shown in the page, so not too small, but also maybe not a full-width (1920x1080) screenshot if it is only shown 600px wide in the end, because it might be illegible. But that is stuff we can test.

**Comment by tutebatti** [comment]
> (2) directly link to an external URL for the image

Ok, we will do that, then. Where would be a good place to host that? Could we just incorporate it via the CMS of the website of our chair at HU?

> Resolution: matching roughly the size it is shown in the page

Alright, but concerning adaptable layouting (small screens of phones etc.), I am not sure what is best practice. Could you give me a rough estimate, say 1200px?

**Comment by mfranke93** [comment]
>> (2) directly link to an external URL for the image
>
> Ok, we will do that, then. Where would be a good place to host that? Could we just incorporate it via the CMS of the website of our chair at HU?

My mistake. I thought the image is already hosted somewhere.

If we are allowed to host it ourselves (at CMS or whatever) anyways, we can host it directly on the server as well. If we do not want it in the repo, the easiest way of accomplishing that is to use the [override feature](https://github.com/UniStuttgart-VISUS/damast/tree/feature/documentation-user-guide#overriding-pages) for the image.

>> Resolution: matching roughly the size it is shown in the page
>
> Alright, but concerning adaptable layouting (small screens of phones etc.), I am not sure what is best practice. Could you give me a rough estimate, say 1200px?

I really am not that familiar with web design for phones. But it should be clear that on a phone, you will not be able to see all details of the screenshots anyways without zooming in. I think modern phones have *effective* (ignoring device pixel ratios) screen widths (portrait) of 400-500px. My suggestion would be to select the picture sizes for desktop, and to ensure that images are just full-width if the screen size is smaller using media queries. It would then be up to phone visitors to zoom into images on their own if they want more detail.

**Comment by tutebatti** [comment]
> If we are allowed to host it ourselves (at CMS or whatever) anyways, we can host it directly on the server as well.

Sounds good.



> My suggestion would be to select the picture sizes for desktop, and to ensure that images are just full-width if the screen size is smaller using media queries. It would then be up to phone visitors to zoom into images on their own if they want more detail.

Alright. So I will assume a resolution of the screen of 1920x1080 but, as advised, stay below that for the picture.

**Comment by tutebatti** [comment]
> * Added an `<aside>` element for the in-page navigation, such that it is always visible.

I'm still not happy with that, simply for aesthetic reasons. If it's implemented as it is now, it conflicts with the symmetric design of the rest. If there's a navigation floating at the side (or anywhere else) it conflicts with the parallax effect in that it is an ugly overlay over the background picture depending on scrolling position. Since Dorothea, Ralph, and also others were happy with the original design, adding a small navigation at the top, right below the first screenshot, I would like to revert to that.

PS: This is markedly different from the visualization documentation, with a much longer text that will need some sort of navigation.

**Comment by mfranke93** [comment]
I see. Of course, the decision is yours to make. Seeing, however, that you are set on your design already, and also that we probably do not have the time to integrate it properly for the initial public release, I would suggest the following: You continue to create and design the start page as a standalone web page separately (i.e., not on the `feature/new-home-page`) branch for now, and can embed that into the instance at HU using the [override feature](https://github.com/UniStuttgart-VISUS/damast/tree/feature/documentation-user-guide#overriding-pages). We can see about integrating that into the repo at a later date, but I think now is not a good time. This also makes it easier for you to incrementally share the versions with the others without me having to deploy them to the testing server each time. This was what the override feature was intended for anyways, alongside with the imprint (Impressum) and GDPR pages (for which you should also provide overrides).

> PS: This is markedly different from the visualization documentation, with a much longer text that will need some sort of navigation.

Yes, I am aware of that. At some time, I wanted to fix that, but now is not the time. I put it in an issue though (#114).

**Comment by tutebatti** [comment]
> Yes, I am aware of that. At some time, I wanted to fix that, but now is not the time.

That's fine. Don't worry about it. What's ugly is only #111.

I will have a look into the override feature and report back, if I have questions.

Thank you!

**Comment by tutebatti** [comment]
> I will have a look into the override feature and report back, if I have questions.

If I understand correctly, the overriding startpage would be completely different from the rest, i.e. not embedded in the `main` element of the current startpage setup. Right? That means, I would need to have an element with a link to the visualization such as "Go to visualization", correct?

**Comment by mfranke93** [comment]
Yes, that would be the entire page. But keep in mind it is still a Jinja2 template. That means it *could* all be valid HTML, but it could also contain Jinja2 syntax. To get the proper URLs to other pages on the website, it is better to use the Jinja2 functions than to hard-code URLs.

Essentially, you are replacing the [default root index.html](https://github.com/UniStuttgart-VISUS/damast/blob/main/dhimmis/root/templates/root/index.html). My suggestion would be to take that as a reference, and to also *derive* from the `base.html` template as it is done in that one. This way, the header and footer would still be there as with the other pages.

But of course, you could also make it an entirely different page.

**Comment by tutebatti** [comment]
> To get the proper URLs to other pages on the website, it is better to use the Jinja2 functions than to hard-code URLs.

I copy.

> Essentially, you are replacing the [default root index.html](https://github.com/UniStuttgart-VISUS/damast/blob/main/dhimmis/root/templates/root/index.html). My suggestion would be to take that as a reference, and to also _derive_ from the [`base.html` template](https://github.com/UniStuttgart-VISUS/damast/blob/main/dhimmis/templates/base.html) as it is done in that one. This way, the header and footer would still be there as with the other pages.

Thanks. I will give my best. :)

**Comment by tutebatti** [comment]
Alright, it seemed more reasonable to me to just modify the index.html and root.scss in the branch `feature/new-home-page`. I hope this is ok.

I commented the `aside` element with the navigation (including editing the css) and added the old navigation that was part of the `main` element.

I also partially changed and added css styles for the map screenshot. There might be one last change regarding this map screenshot that I'm still trying to figure out. Once this is done, I would like to ask you, @mfranke93, to have a look at the final result. (I.e., not now.)

Other than that, there will be the style `font-variant: small-caps;` to add once the content is ready to be included in the source text as well and we should be good to go for the initial public version.

**Comment by mfranke93** [comment]
> Alright, it seemed more reasonable to me to just modify the index.html and root.scss in the branch feature/new-home-page. I hope this is ok.

I am planning on deleting (or at least not merging) that branch. My understanding was that the changes I suggested to your design in that branch were not what you wanted, and that you would just continue working on your standalone HTML design for now. Unless you have got a development server running on your end by now, that would also be the best way to actually *see* the changes you make (especially the SCSS). Of course you can copy the `index.html` from that branch somewhere for editing, but I am not sure if it wouldn't be better to continue editing with a version without my changes in it.

**Comment by tutebatti** [comment]
I will have look at what the changes are comparing the `main` branch and the `feature/new-home-page` branch.

It is not as easy for me to overlook the entire workings with the different folders and the jinja2 code; that is the main reason why changing what you implemented based on my design is easier for me than coming up with something new.

**Comment by tutebatti** [comment]
What should I put into an html in the `override/templates` to take care of this `#dynamic_date as, e.g., April 6, 2016]#`. :see_no_evil: ?

**Comment by mfranke93** [comment]
Do you want this to be the current date? My understanding until now was that this would be written as such, for the visitor to fill in.

**Comment by tutebatti** [comment]
It should be the date of the individual day, if possible.

**Comment by mfranke93** [comment]
In the format `full month` `space` `day of month without leading zero` `comma` `space` `full year`?

**Comment by tutebatti** [comment]
Exactly.

**Comment by mfranke93** [comment]
Put in `{{ today }}`

It won't show anything yet, but will as soon as 4acddd153937ac6fd0c36d5b9b35d1ab1ae0cd83 is merged and deployed with `release/v1.1.1`.

**Comment by tutebatti** [comment]
Thank you very much!

**Comment by tutebatti** [comment]
I am not sure how you, @mfranke93, want to handle the default appearance "shipped" with the github repo, but as far as the public version is concerned, this can be closed.

**Comment by mfranke93** [comment]
I think the plan was to incorporate the new design into the repo, but to do that after the publication deadline and include it as an override for now, right?

Whether you want it in the repo is up to you, but if anything changes in the repo that breaks the current override pages, it would be up to you to fix it on your end if the start page is not in the repo.
